torch/distributed/_sharded_tensor/__init__.py

```diff
@@ -596,7 +591,9 @@ def _reshard_output(
         A :class:`torch.nn.Module` object with collection API hooked.
     """
     def hook_func(_module, _input, output):
-        return output.reshard(resharding_spec)
+        if isinstance(output, ShardedTensor) or isinstance(output, _PartialTensor):
```

Comment:
Output can probably be a tuple too? I guess we should handle that case and reshard all elements of the tuple that are ShardedTensor or PartialTensor. We should also update the docs to mention that we reshard only ShardedTensor and PartialTensor.

Reply:
Oh, I see. I have not thought about the case of tuple. Maybe we need to check if it's an Iterable?

Reply:
Filed an issue here to follow up: https://github.com/pytorch/pytorch/issues/72022.